### PR TITLE
Issue 199 nmatch enhancement

### PR DESCRIPTION
**Pull request by Rita-C**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -11,12 +11,16 @@ buildscript {
...
     }
 }
+//
+//plugins {
+//    id "com.gorylenko.gradle-git-properties" version "2.3.1"
+//}
 
 plugins {
-    id "com.gorylenko.gradle-git-properties" version "2.3.1"
+    id "com.gorylenko.gradle-git-properties" version "2.4.0-rc1"
```
Comment (nickdos) [inline build.gradle:20]: Is a release candidate allowed for prod? Would prefer and stable release if there is a version... if not then that's fine to leave as `-rc1`.

Comment (Rita-C) [reply build.gradle:20]: I'll revert it back, had to do it to start the app locally.

Comment (nickdos) [reply build.gradle:20]: Looks like 2.3.1 is last stable (June 2021) - is it not Grails 4 compatible?

**Review by nickdos — APPROVED** [review]

**Review by nickdos — APPROVED** [review]

File: grails-app/services/au/org/ala/specieslist/HelperService.groovy [hunk 2]
```diff
@@ -596,6 +685,27 @@ class HelperService {
         }
     }
 
+    def  matchValuesToSpeciesListItem(String[] values, Map termIndex, SpeciesListItem sli){
+        String rawScientificName = termIndex.containsKey(RAW_SCIENTIFIC_NAME) ? values[termIndex[RAW_SCIENTIFIC_NAME]] : null
```
Comment (charvolant) [inline grails-app/services/au/org/ala/specieslist/HelperService.groovy:689]: Perhaps make this a closure? def get = { key -> termIndex.containsKey[key] ? values[termIndex[key] : null } and then String family = get(FAMILY)

**Review by charvolant — APPROVED** [review]

File: grails-app/services/au/org/ala/specieslist/HelperService.groovy (hunk 2 above)
Comment (Rita-C) [reply grails-app/services/au/org/ala/specieslist/HelperService.groovy:689]: closure looks smart but hard to debug.

**Pull request merged**